### PR TITLE
Update dlc before release

### PR DESCRIPTION
**Pull request by nkohen**
Built on top of #2516

### CODE REVIEW

**Comment by benthecarman** [comment]
Has rebase issues

File: app/gui/src/main/scala/org/bitcoins/gui/dlc/dialog/ViewDLCDialog.scala [hunk 1]
```diff
@@ -185,20 +180,27 @@ object ViewDLCDialog {
...
+                                       roundingIntervals) =>
           val previewGraphButton: Button = new Button("Preview Graph") {
             onAction = _ => {
+              val payoutCurve = if (status.isInitiator) {
+                outcomeValueFunc
+              } else {
+                DLCPayoutCurve(outcomeValueFunc.points.map { point =>
+                  point.copy(payout =
+                    status.totalCollateral.satoshis.toLong - point.payout)
+                })
```
Comment (benthecarman) [inline app/gui/src/main/scala/org/bitcoins/gui/dlc/dialog/ViewDLCDialog.scala]: this is my bad but can use `NumericContractDescriptor.flip` here

File: core/src/main/scala/org/bitcoins/core/protocol/dlc/OracleInfo.scala [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+}
+
+/** Specifies a single oracles' information for an Enumerated Outcome DLC
+  * through an announcement
+  */
+case class EnumSingleOracleInfo(announcement: OracleAnnouncementTLV)
+    extends SingleOracleInfo
+    with EnumOracleInfo {
+  require(announcement.eventTLV.eventDescriptor
+            .isInstanceOf[EnumEventDescriptorV0TLV],
```
Comment (benthecarman) [inline core/src/main/scala/org/bitcoins/core/protocol/dlc/OracleInfo.scala:111]: Maybe eventually we will want to make things like `OracleAnnouncementWithEnumTLV` so we don't need these

**Comment by benthecarman** [comment]
Otherwise LGTM

**Review by benthecarman — APPROVED** [review]

File: app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCTableView.scala [hunk 3]
```diff
@@ -68,18 +69,26 @@ class DLCTableView(model: DLCPaneModel) {
       text = "Oracle"
       prefWidth = 150
       cellValueFactory = { status =>
-        new StringProperty(status, "Oracle", status.value.oracleInfo.pubKey.hex)
+        new StringProperty(
+          status,
+          "Oracle",
+          status.value.oracleInfo.asInstanceOf[SingleOracleInfo].publicKey.hex
+        ) // FIXME
```
Comment (Christewart) [inline app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCTableView.scala:76]: ping

File: app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCTableView.scala [hunk 4]
```diff
@@ -68,18 +69,26 @@ class DLCTableView(model: DLCPaneModel) {
...
-          "Event",
-          status.value.oracleInfo.nonces.map(_.hex).mkString(""))
+        new StringProperty(status,
+                           "Event",
+                           status.value.oracleInfo
+                             .asInstanceOf[SingleOracleInfo]
+                             .nonces
+                             .map(_.hex)
+                             .mkString("")
+        ) // FIXME
```
Comment (Christewart) [inline app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCTableView.scala:91]: ping

File: core/src/main/scala/org/bitcoins/core/protocol/dlc/CETCalculator.scala [hunk 5]
```diff
@@ -402,4 +405,381 @@ object CETCalculator {
...
+    * @param minFailExp The exponent (of 2) representing the difference up to
+    *                   which support is guaranteed
+    */
+  def computeCoveringCETsBinary(
+      numDigits: Int,
+      cetDigits: Vector[Int],
+      maxErrorExp: Int,
+      minFailExp: Int,
+      maximizeCoverage: Boolean,
+      numOracles: Int): Vector[Vector[Vector[Int]]] = {
```
Comment (Christewart) [inline core/src/main/scala/org/bitcoins/core/protocol/dlc/CETCalculator.scala]: So can we get some types for this stuff? I can't be the only one to think `Vector[Vector[Vector[Int]]]` is unreasonable to try and understand.

File: core/src/main/scala/org/bitcoins/core/protocol/dlc/CETCalculator.scala [hunk 6]
```diff
@@ -402,4 +405,381 @@ object CETCalculator {
...
+    */
+  def computeMultiOracleCETsBinary(
+      numDigits: Int,
+      function: DLCPayoutCurve,
+      totalCollateral: Satoshis,
+      rounding: RoundingIntervals,
+      maxErrorExp: Int,
+      minFailExp: Int,
+      maximizeCoverage: Boolean,
+      numOracles: Int): Vector[(Vector[Vector[Int]], Satoshis)] = {
```
Comment (Christewart) [inline core/src/main/scala/org/bitcoins/core/protocol/dlc/CETCalculator.scala]: >Vector[(Vector[Vector[Int]], Satoshis)] 

is also really hard to understand

**Review by Christewart — CHANGES_REQUESTED** [review]
So this is going to take multiple rounds or review, but I do feel strongly about getting more structure around `CETCalculator`. There is a lot of `Vector[Vector[]]` or even `Vector[Vector[Vector[]]]`.

It would really help me code review if these had useful names (and can we apply to invariants to any of these things??) to better understand what is happening.

File: app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCTableView.scala (hunk 3 above)
Comment (nkohen) [reply app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCTableView.scala:76]: cc @benthecarman

File: app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCTableView.scala (hunk 4 above)
Comment (nkohen) [reply app/gui/src/main/scala/org/bitcoins/gui/dlc/DLCTableView.scala:91]: cc @benthecarman

File: core/src/main/scala/org/bitcoins/core/protocol/dlc/CETCalculator.scala (hunk 5 above)
Comment (nkohen) [reply core/src/main/scala/org/bitcoins/core/protocol/dlc/CETCalculator.scala]: Better?

File: core/src/main/scala/org/bitcoins/core/protocol/dlc/CETCalculator.scala (hunk 6 above)
Comment (nkohen) [reply core/src/main/scala/org/bitcoins/core/protocol/dlc/CETCalculator.scala]: Better?

File: testkit/src/main/scala/org/bitcoins/testkit/core/gen/TLVGen.scala [hunk 7]
```diff
@@ -144,15 +145,17 @@ trait TLVGen {
...
     }
 
-    def genContractInfos(outcomes: Vector[String], totalInput: CurrencyUnit): (
-        SingleNonceContractInfo,
-        SingleNonceContractInfo) = {
+    def genContractDescriptors(
+        outcomes: Vector[String],
+        totalInput: CurrencyUnit): (
+        EnumContractDescriptor,
+        EnumContractDescriptor) = {
```
Comment (Christewart) [inline testkit/src/main/scala/org/bitcoins/testkit/core/gen/TLVGen.scala:152]: nit: return type

**Review by Christewart — APPROVED** [review]

**Pull request merged**